instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java

```diff
@@ -40,6 +42,12 @@ public static void onEnter(
     }
     HttpServletRequest httpServletRequest = (HttpServletRequest) request;
 
+    String snippet = ExperimentalSnippetHolder.getSnippet();
+    if (!snippet.isEmpty()
+        && !((HttpServletResponse) response)
+            .containsHeader(SnippetInjectingResponseWrapper.FAKE_SNIPPET_HEADER)) {
+      response = new SnippetInjectingResponseWrapper((HttpServletResponse) response, snippet);
```

Comment:
@mateuszrzeszutek great memory, I forgot about that 😅

@siyuniu-ms I found the issue @mateuszrzeszutek is referencing: #1156

> though that's material for next PRs, I guess

👍 that sounds good since this is a big PR and the wrapping only happens when opting-in to snippet injection

@siyuniu-ms can you open an issue in this repository to track this?

Reply:
#8267